src/cpp/database/database.cpp

```diff
@@ -279,6 +280,33 @@ EntityId Database::insert_new_topic(
     return entity_id;
 }
 
+bool Database::is_type_in_database(
+        const std::string& type_name)
+{
+
+    if (type_idls_.find(type_name) != type_idls_.end())
+    {
+        return true;
+    } 
+    else
+    {
+        return false;
+    }
+}
+
+void Database::insert_new_type_idl(
+            const std::string& type_name,
+            const std::string& type_idl)
+{
+    std::lock_guard<std::shared_timed_mutex> guard(mutex_);
+    if (type_name.empty() || type_idl.empty())
+    {
+        throw BadParameter("Type name or type IDL cannot be empty");
+    }
+    type_idls_[type_name] = type_idl;
+}
+
+
```

Comment:
Remove extra line break
```suggestion
}

```